recipes-bsp/xt-rpi-u-boot-scr/files/boot.cmd.xen.in

```diff
@@ -0,0 +1,17 @@
+setenv xen_addr_r 0x02700000
+setenv xenpolicy_addr_r 0x02900000
+fatload @@BOOT_MEDIA@@ 0 ${kernel_addr_r} @@DOM0_IMAGE@@
+fdt addr ${fdt_addr} 0x40000
+
+fdt resize
+fdt chosen
+fdt set /chosen/modules/module@1 reg <${kernel_addr_r} 0x${filesize} >
+fatload @@BOOT_MEDIA@@ 0 ${xenpolicy_addr_r} xenpolicy
+fdt set /chosen/modules/module@2 reg <${xenpolicy_addr_r} 0x${filesize} >
+fdt set /chosen bootargs "dom0_mem=128M console=dtuart dtuart=/soc/serial@7d001000 dom0_max_vcpus=2 loglvl=all xsm=flask flask=permissive bootscrub=1"
+fdt rm /chosen/stdout-path
+fdt print /chosen
```

Comment:
I think you don't need all of this chosen manipulations. All this can be done during dtb building.

Reply:
I have to set /chosen/bootargs property because Rpi bootloader changes it.
In /chosen/modules/module@*/reg property I put actual file size

Reply:
Hm. didn't know that. print should definitely be removed. otherwise it looks good. because approach when replacing fdt in u-boot will be unneccessary step.

Reply:
no. it is not. At least not now.

Reply:
Yes. We don't know file sizes before build. This script inserts actual file sizes on run time.